pkg/diagnostics/http_monitoring.go

```diff
@@ -56,14 +67,22 @@ type httpMetrics struct {
 	clientRoundtripLatency *stats.Float64Measure
 	clientCompletedCount   *stats.Int64Measure
 
-	healthProbeCompletedCount  *stats.Int64Measure
-	healthProbeRoundripLatency *stats.Float64Measure
+	healthProbeCompletedCount   *stats.Int64Measure
+	healthProbeRoundTripLatency *stats.Float64Measure
 
 	appID   string
 	enabled bool
 
 	// Enable legacy metrics, which includes the full path
 	legacy bool
+
+	pathMatching pathMatchingConfig
+}
+
+type pathMatchingConfig struct {
+	enabled           bool
+	virtualIngressMux *http.ServeMux
+	virtualEgressMux  *http.ServeMux
```

Comment:
Create a `patchMaching` struct containing only a `mux *http.ServeMux`. `httpMetrics` can then have 2 `patchMatching`s, ingress and egress.